app/controllers/payments_controller.rb

```diff
@@ -19,8 +19,8 @@ def index
         input_params_month = DateTime.now.month.to_s
       end
       i = DateTime.parse("#{input_params_year}-#{input_params_month}-01")
-      payments = Payment.where(input_date: [i..i + 1.month - 1.day])
-      payments.each do |payment|
+      search_area = current_user.payments.where(input_date: [i..i + 1.month - 1.day])
```

Comment:
`@payments` にはログインしているユーザーの情報で絞られた情報が入っていると思うのでそれを活用しましょう。


```suggestion
      search_area = @payments.where(input_date: [i..i + 1.month - 1.day])
```